Docs updates reflecting hook retry/backoff changes (#8490)
habitat-sh/habitat: components/docs-chef-io/content/habitat/sup_config.md

### track-channel: Always run what is at the head of a given channel. This enables service rollback where demoting a package from a channel will cause the package to rollback to an older version of the package. A ramification of enabling this condition is packages newer than the package at the head of the channel will be automatically uninstalled during a service rollback.
update_condition = "track-channel"

### The minimum duration in seconds to wait before restarting a service due to a init / run hook failure.
###
### The default value if this config is absent is 0 seconds to preserve legacy supervisor behavior.
service_min_backoff_period = 10

### The maximum duration in seconds to wait before restarting a service due to init / run hook failure.
###
### The default value if this config is absent is 0 seconds to preserve legacy supervisor behavior.
service_max_backoff_period = 180

### The duration of time in seconds to wait before resetting the current backoff duration to the 'service_min_backoff_period'.

> ### This is important to ensure the supervisor does not mistake a slow start with a service failure.

Reviewer: The service considered successfully started once the cooldown has elapsed. This may be a bit confusing but I'll try to list the various cases to make it clear.
- Service Takes 30s to start
  - Cooldown is 15s
    - Service considered success after 15s even though it hasn't started up and could potentially fail later in the startup process. Here a slow start is confused with a successful startup.
  - Cooldown is 60s
    - Service considered success correctly after 60s. Here any failure before 60s is correctly identified as failure, and we do not reset the backoff period.

Namely, there is no scenario where we mistake a slow start for a service failure. We always identify failures correctly. But we mistakenly assume success for cases which are actually failures. 

We need correct identification of success in order to reset the backoff period and nothing else.

Notice that even if the cooldown was 600 secs it would only prevent the backoff from resetting, potentially having the service restarted continuously between the min and the capped max backoff period. All failures would still be detected correctly and cause some backoff to be applied.

### track-channel: Always run what is at the head of a given channel. This enables service rollback where demoting a package from a channel will cause the package to rollback to an older version of the package. A ramification of enabling this condition is packages newer than the package at the head of the channel will be automatically uninstalled during a service rollback.
update_condition = "track-channel"

### The minimum duration in seconds to wait before restarting a service due to a init / run hook failure. 
### 
### The default value if this config is absent is 0 seconds to preserve legacy supervisor behavior. 
service_min_backoff_period = 10 

### The maximum duration in seconds to wait before restarting a service due to init / run hook failure. 
### 
### The default value if this config is absent is 0 seconds to preserve legacy supervisor behavior. 
service_max_backoff_period = 180 

### The duration of time in seconds to wait before resetting the current backoff duration to the 'service_min_backoff_period'. 

> ### This is important to ensure the supervisor does not mistake a slow start for a successful startup. 

Reviewer 2: Should this be more like: 
```suggestion
### This is important to ensure the supervisor does not mistake a slow start with a service failure. 
```
